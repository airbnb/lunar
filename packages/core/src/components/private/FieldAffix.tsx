import withStyles from '../../composers/withStyles';
import BaseAffix, { Props as BaseProps } from './BaseAffix';

export type Props = BaseProps;

export default withStyles(({ color, font, unit, ui, pattern }) => ({
  affix: {
    ...font.textRegular,
    height: '100%',
    display: 'flex',
    alignItems: 'center',
    border: ui.borderThick,
    borderRadius: ui.borderRadius,
    background: color.accent.bgHover,
    paddingLeft: unit * 1.5,
    paddingRight: unit * 1.5,
  },

  affix_compact: {
    ...font.textSmall,
    paddingLeft: unit,
    paddingRight: unit,
  },

  affix_after: {
    borderLeft: 0,
    borderTopLeftRadius: 0,
    borderBottomLeftRadius: 0,
  },

  affix_before: {
    borderRight: 0,
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
  },

  affix_disabled: {
    ...pattern.disabled,
  },
}))(BaseAffix);
