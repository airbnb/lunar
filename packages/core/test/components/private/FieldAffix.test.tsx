import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import FieldAffix from '../../../src/components/private/FieldAffix';

describe('<FieldAffix />', () => {
  it('renders before and after with different classes', () => {
    const wrapper = shallowWithStyles(<FieldAffix before>Child</FieldAffix>);
    const beforeClass = wrapper.prop('className');

    wrapper.setProps({
      before: false,
      after: true,
    });

    expect(wrapper.prop('className')).not.toBe(beforeClass);
  });

  it('renders compact', () => {
    const wrapper = shallowWithStyles(
      <FieldAffix after compact>
        Child
      </FieldAffix>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('errors if both props used', () => {
    expect(() => {
      shallowWithStyles(
        <FieldAffix before after>
          Child
        </FieldAffix>,
      );
    }).toThrowError();
  });
});
