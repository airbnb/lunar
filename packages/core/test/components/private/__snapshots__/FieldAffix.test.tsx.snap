// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldAffix /> renders compact 1`] = `
<div
  className="affix_t14f0p-o_O-affix_compact_ym0g6v-o_O-affix_after_h7uxoa"
>
  Child
</div>
`;
